VideoLocker/src/main/java/org/edx/mobile/util/images/TopAnchorWidthFillTransformation.java

```diff
@@ -0,0 +1,49 @@
+package org.edx.mobile.util.images;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.Paint;
+import android.os.Build;
+
+import com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool;
+import com.bumptech.glide.load.resource.bitmap.BitmapTransformation;
+
+/**
+ * Created by miankhalid on 9/9/15.
+ */
+public class TopAnchorWidthFillTransformation extends BitmapTransformation {
+    private final Paint paint = new Paint(Paint.FILTER_BITMAP_FLAG);
+
+    public TopAnchorWidthFillTransformation(Context context) {
+        super(context);
+    }
+
+    @Override
+    protected Bitmap transform(BitmapPool pool, Bitmap toTransform,
+                               int outWidth, int outHeight) {
+        final int width = toTransform.getWidth();
+        final float widthRatio = outWidth / (float) width;
+        if (outWidth > width) {
```

Comment:
This was an optimization to just match the aspect ratio of the `View` where the `Bitmap` has a smaller width, and not artificially scale up the `Bitmap` (as `ImageView` will automatically scale it up to match it's coordinates unless explicitly set not to do so). However, Glide uses a `TransitionDrawable` to transition from the placeholder to the actual image, and as we can't guarantee that the placeholder would also have a matching aspect ratio, this can cause the scaling to not be performed properly to fill the `View`.

Setting the `scaleType` of the `ImageView` items to `fitXY` would take care of this issue, but we might not always want to do that for the placeholders. Therefore, we need to get rid of this optimization, by removing this `if` section, and scale the `Bitmap` in all cases to match the exact dimensions of the `View` (so remove the `if (outWidth < width)` condition from the scaling).